Change for GNOME 3 where gnome-terminal expects a UTF-8 locale - not suitable for upstream

--- tigervnc-1.10.1/unix/vncserver
+++ tigervnc-1.10.1/unix/vncserver
@@ -33,6 +33,15 @@ if($slashndx>=0) {
     $exedir = substr($0, 0, $slashndx+1);
 }
 
+# With Gnome 3, gnome-terminal-server won't exec unless vncserver is started with UTF-8 locale.
+for ($ENV{LANG}) {
+    /\.UTF-8/     and do {last;};
+    /^C$|^POSIX$/ and do {$ENV{LANG} = "C.UTF-8"; last;};
+    /\..*\@/      and do {$ENV{LANG} =~ s/\..*\@/.UTF-8\@/g; last;};
+    /\./          and do {$ENV{LANG} =~ s/\..*/.UTF-8/g; last;};
+    /_/           and do {$ENV{LANG} = "$ENV{LANG}.UTF-8"; last;};
+    $ENV{LANG} = "C.UTF-8";
+}
 &SanityCheck();
 
 #
